zookeeper-server/src/test/java/org/apache/zookeeper/common/ZKConfigTest.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zookeeper.common;
+
+
+import static org.junit.Assert.assertEquals;
+import java.io.File;
+import java.util.concurrent.TimeUnit;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+
+
+
+public class ZKConfigTest {
+
+    private static final File testData = new File(System.getProperty("test.data.dir", "src/test/resources/data"));
+    X509Util x509Util;
+    @Rule
+    public Timeout timeout = new Timeout(10, TimeUnit.SECONDS);
+
+    @BeforeClass
+    public static void init() {
+        if (!testData.exists()) {
+            testData.mkdirs();
+        }
+    }
+
+    // property is not set we should get the default value
+    @Test
+    public void testBooleanRetrievalFromPropertyDefault() {
+        ZKConfig conf = new ZKConfig();
+        String prop = "UnSetProperty" + System.currentTimeMillis();
+        boolean defaultValue = false;
+        boolean result = conf.getBoolean(prop, defaultValue);
+        assertEquals(defaultValue, result);
+    }
+
+    // property is set to an valid boolean, we should get the set value
+    @Test
+    public void testBooleanRetrievalFromProperty() {
+        boolean value = true;
+        boolean defaultValue = false;
+        x509Util = new ClientX509Util();
+        System.setProperty(x509Util.getSslProtocolProperty(), Boolean.toString(value));
+        ZKConfig conf = new ZKConfig();
+        boolean result = conf.getBoolean(x509Util.getSslProtocolProperty(), defaultValue);
+        assertEquals(value, result);
+        System.clearProperty(x509Util.getSslProtocolProperty());
```

Comment:
Clear properties in tearDown method
@After    
public void tearDown() throws Exception {       }

Reply:
Done, thanks